webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,398 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as md from "./marked";
+import * as compiler from './compiler';
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+export interface SnippetBuilderState {
+    visible?: boolean;
+    tsOutput?: string;
+    projectView?: pxt.editor.IProjectView;
```

Comment:
does not belong to the state

Reply:
this doesn't just capture the final output, but also the intermediate output as you're working through questions. How else should this be stored?

Reply:
It is stored in ISettingsProps as the "parent" field.

Reply:
I'm now passing this down as the parent prop.